research-app/src/SpreadsheetItem.vue

```diff
@@ -158,47 +169,64 @@ export default class CatalogItem extends Vue {
     this.$options.computed = {
       ...mapState(wwtResearchAppNamespace, {
         visible: (_state, getters) =>
-          getters["researchAppHipsCatalogVisibility"](this.catalog),
+          getters["researchAppTableLayerVisibility"](this.layer),
       }),
-      ...mapGetters(wwtEngineNamespace, ["spreadsheetStateForHipsCatalog"]),
+      ...mapGetters(wwtEngineNamespace, [
+        "spreadsheetState",
+        "spreadsheetStateForHipsCatalog"
+      ]),
       ...this.$options.computed,
     };
 
     this.$options.methods = {
       ...this.$options.methods,
       ...mapMutations(wwtEngineNamespace, [
         "applyTableLayerSettings",
+        "deleteLayer",
         "removeCatalogHipsByName",
       ]),
       ...mapMutations(wwtResearchAppNamespace, [
-        "removeResearchAppCatalogHips",
-        "setResearchAppCatalogHipsVisibility",
+        "removeResearchAppTableLayer",
+        "setResearchAppTableLayerVisibility",
       ]),
     };
   }
 
   // Tied to the store value
   visible!: boolean;
 
+  spreadsheetState!: (id: string) => SpreadSheetLayerSettingsInterfaceRO | null;
   spreadsheetStateForHipsCatalog!: (
     _n: string
   ) => SpreadSheetLayerSettingsInterfaceRO | null;
 
   applyTableLayerSettings!: (_o: ApplyTableLayerSettingsOptions) => void;
+  deleteLayer!: (id: string | Guid) => void;
   removeCatalogHipsByName!: (name: string) => void;
-  removeResearchAppCatalogHips!: (catalog: ImagesetInfo) => void;
-  setResearchAppCatalogHipsVisibility!: (args: {
-    catalog: ImagesetInfo;
+  removeResearchAppTableLayer!: (layer: LayerInfo) => void;
+  setResearchAppTableLayerVisibility!: (args: {
+    layer: LayerInfo;
     visible: boolean;
   }) => void;
 
   // Local state
 
+  editing = false;
   hasFocus = false;
   isSelected = false;
+  isLayerHips = false;
+
+  layerId(): string {
+    return isImageSetInfo(this.layer) ? this.layer.name : this.layer.id;
+  }
+
+  layerState(): SpreadSheetLayerSettingsInterfaceRO | null {
+    const id = this.layerId();
+    return isImageSetInfo(this.layer) ? this.spreadsheetStateForHipsCatalog(id) : this.spreadsheetState(id);
```

Comment:
This is another area where maybe it's worthwhile to bring this logic closer to the definition of (what's currently named) the LayerInfo type.